test/go.mod

```
@@ -0,0 +1,142 @@
+module sigs.k8s.io/cluster-api-provider-vsphere/test
+
+go 1.20
+
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.6.0
+
+replace sigs.k8s.io/cluster-api/test => sigs.k8s.io/cluster-api/test v1.6.0
```

Let's bump a bit more to fix https://github.com/kubernetes-sigs/cluster-api/issues/9775
```suggestion
replace sigs.k8s.io/cluster-api/test => sigs.k8s.io/cluster-api/test v1.6.1-0.20231212113133-a1fff03d13f3
```

---

(to pick up a fix from core CAPI e2e tests)

---

@fabriziopandini I think this one is missing